adam_preprocessing/shape_stroke_extraction.py

```diff
@@ -546,41 +563,52 @@ def get_strokes(self):
             )
 
         # Calculate relative sizes using already-calculated absolute sizes:
-        for i in range(len(data)):
+        for i in range(len(objects)):
             relative_size = {'x_bigger_than': [], 'y_bigger_than': []}
-            for j in range(len(data)):
+            for j in range(len(objects)):
                 if i == j:
                     continue
                 else:
-                    other_object_name = data[j]['object_name']
+                    other_object_name = objects[j]['object_name']
                     for axis in ('x', 'y'):
                         # Relative size is a discrete ('bigger_than') relation
-                        if data[i]['size'][axis] > \
-                                data[j]['size'][axis]:
+                        if objects[i]['size'][axis] > \
+                                objects[j]['size'][axis]:
                             relative_size[f'{axis}_bigger_than'].append(other_object_name)
 
                         # Normalize relative distances by the relevant axis
                         # sizes of both objects
-                        object_axis_size = data[i]['size'][axis]
-                        other_object_axis_size = data[j]['size'][axis]
-                        data[i]['relative_distance'][other_object_name][f'{axis}_offset'] /= (object_axis_size * other_object_axis_size)
+                        object_axis_size = objects[i]['size'][axis]
+                        other_object_axis_size = objects[j]['size'][axis]
+                        objects[i]['relative_distance'][other_object_name][f'{axis}_offset'] /= (object_axis_size * other_object_axis_size)
 
                     # Normalize euclidean distance by product of areas of both
                     # objects
-                    object_area = data[i]['size']['area']
-                    other_object_area = data[j]['size']['area']
-                    data[i]['relative_distance'][other_object_name]['euclidean'] /= (object_area * other_object_area)
-            data[i]['relative_size'] = relative_size
-
-        self.data = data
+                    object_area = objects[i]['size']['area']
+                    other_object_area = objects[j]['size']['area']
+                    objects[i]['relative_distance'][other_object_name]['euclidean'] /= (object_area * other_object_area)
+            objects[i]['relative_size'] = relative_size
+
+        # Determine touching relations based on closest-point distance
+        touching = set()
```

Comment:
I'd prefer to use a list for `touching` here. Sets aren't deterministically ordered and I expect this to result in inconsistent serializations. Meanwhile, we don't actually need any set operations -- the only operations used on `touching` are (1) append, and (2) YAML serialization.